woocommerce/google-listings-and-ads
# Use `/mc/setup` endpoint and remove `gla_setup_mc_saved_step` option

### Changes proposed in this Pull Request:

Closes #454 . Uses API created from https://github.com/woocommerce/google-listings-and-ads/pull/443.

This PR calls the `/mc/setup` endpoint in the Setup MC flow. The result will be used to display the correct step to the users. 

This PR also removes the use of `gla_setup_mc_saved_step` options and the related `useGetOptions` hook.

### Screenshots:

No UI changes.

### Detailed test instructions:

Pre-requisite: disconnect all accounts to start Setup MC flow from scratch.

1. Open https://gla1.loca.lt/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc
2. You should see Step 1 displayed to you.
3. Click on the Step 2 and Step 3 at the top. You should not be able to proceed to them.
4. Connect the accounts in Step 1.
5. Click the Continue button at the bottom of Step 1. You should now see Step 2.
6. Click on the Step 1 at the top. You should go back to Step 1.
7. Click on the Step 2 at the top. You should go back to Step 2.
8. Click on the Step 3 at the top. You should not be able to go to Step 3.
9. Fill in the fields in Step 2, and click Continue button at the bottom. You should now see Step 3. 
10. Click on the Step 1 at the top. You should go back to Step 1.
11. Click on the Step 2 at the top. You should go back to Step 2.
12. Click on the Step 3 at the top. You should go back to Step 3.
13. Click on the Complete setup button at the bottom. The page should eventually reload and display a success modal.
14. Open https://gla1.loca.lt/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc again, the page should detect you've completed MC Setup and automatically direct you to the dashboard page.

### Changelog Note:

Use `mc/setup` endpoint and remove the use of `gla_setup_mc_saved_step` option.
